pkg/querier/blocks_store_queryable.go

```diff
@@ -781,28 +813,60 @@ func (q *blocksStoreQuerier) fetchSeriesFromStores(ctx context.Context, sp *stor
 				if s := resp.GetStats(); s != nil {
 					indexBytesFetched += s.FetchedIndexBytes
 				}
-			}
 
-			numSeries := len(mySeries)
-			chunksFetched, chunkBytes := countChunksAndBytes(mySeries...)
+				if ss := resp.GetStreamingSeries(); ss != nil {
+					for _, s := range ss.Series {
+						// Add series fingerprint to query limiter; will return error if we are over the limit
+						limitErr := queryLimiter.AddSeries(s.Labels)
+						if limitErr != nil {
+							return validation.LimitError(limitErr.Error())
+						}
+					}
+					myStreamingSeries = append(myStreamingSeries, ss.Series...)
+					if ss.IsEndOfSeriesStream {
+						// We expect "end of stream" to be sent after the hints and the stats have been sent.
+						break
+					}
+				}
+			}
 
-			reqStats.AddFetchedSeries(uint64(numSeries))
-			reqStats.AddFetchedChunkBytes(uint64(chunkBytes))
-			reqStats.AddFetchedChunks(uint64(chunksFetched))
 			reqStats.AddFetchedIndexBytes(indexBytesFetched)
-
-			level.Debug(spanLog).Log("msg", "received series from store-gateway",
-				"instance", c.RemoteAddress(),
-				"fetched series", numSeries,
-				"fetched chunk bytes", chunkBytes,
-				"fetched chunks", chunksFetched,
-				"fetched index bytes", indexBytesFetched,
-				"requested blocks", strings.Join(convertULIDsToString(blockIDs), " "),
-				"queried blocks", strings.Join(convertULIDsToString(myQueriedBlocks), " "))
+			var streamReader *storeGatewayStreamReader
+			if len(mySeries) > 0 {
+				chunksFetched, chunkBytes := countChunksAndBytes(mySeries...)
+
+				reqStats.AddFetchedSeries(uint64(len(mySeries)))
+				reqStats.AddFetchedChunkBytes(uint64(chunkBytes))
+				reqStats.AddFetchedChunks(uint64(chunksFetched))
+
+				level.Debug(spanLog).Log("msg", "received series from store-gateway",
+					"instance", c.RemoteAddress(),
```

Comment:
Could the instance be a tag on the span so that we don't need to log it on the event?

Reply:
It is one `spanLog` shared between all storegw clients. To have it as a tag, I think we will need to create a spanLog per storegw client. Would that be preferred?

Reply:
That makes sense to me, but @dimitarvdimitrov knows more about the store-gateway than I do - what do you think @dimitarvdimitrov?

Reply:
This seems like a tracing question, it shouldn't have any functional difference, right?

My personal preference is for a single span because in requests with a lot of blocks, the trace view in Grafana becomes difficult to navigate due to too many spans, even though it's technically the more correct way of nesting spans. 

But I don't feel strongly about this.

Reply:
Let's leave it as-is for now, we can change this in a follow-up PR if need be.